src/lib/comms/include/sol-http.h

```diff
@@ -45,15 +45,49 @@ extern "C" {
  * sol_http_client_request() or sol_http_client_request_with_interface()
  */
 enum sol_http_method {
+    /**
+     * Requests data from a specified resource
+     */
     SOL_HTTP_METHOD_GET,
+    /**
+     * Submits data to be processed to a specified resource
+     */
     SOL_HTTP_METHOD_POST,
+    /**
+     * Same as GET, but transfers the status line and
+     * header section only
+     */
     SOL_HTTP_METHOD_HEAD,
+    /**
+     * Removes all current representations of the target
+     * resource given by a URI
+     */
     SOL_HTTP_METHOD_DELETE,
+    /**
+     * Replaces all current representations of the target
+     * resource with the uploaded content
+     */
     SOL_HTTP_METHOD_PUT,
+    /**
+     * Establishes a tunnel to the server identified by a given URI
+     */
     SOL_HTTP_METHOD_CONNECT,
+    /**
+     * Describes the communication options for the target resource
+     */
     SOL_HTTP_METHOD_OPTIONS,
+    /**
+     * Performs a message loop-back test along the path to the
+     * target resource
+     */
     SOL_HTTP_METHOD_TRACE,
+    /**
+     * Used to update partial resources
+     */
     SOL_HTTP_METHOD_PATCH,
+    /**
+     * Invalid input method
```

Comment:
What do you about "client makes an HTTP request by using an HTTP method that does not comply with the HTTP specifications"?